src/Features/Core/Portable/GenerateConstructorFromMembers/AbstractGenerateConstructorFromMembersCodeRefactoringProvider.cs

```diff
@@ -47,60 +49,149 @@ protected AbstractGenerateConstructorFromMembersCodeRefactoringProvider() : this
         protected abstract bool PrefersThrowExpression(DocumentOptionSet options);
 
-        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        public override Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            return ComputeRefactoringsAsync(context.Document, context.Span,
+                (action, applicableToSpan) => context.RegisterRefactoring(action, applicableToSpan),
+                (actions) => context.RegisterRefactorings(actions), context.CancellationToken);
+        }
+
+        public async Task<ImmutableArray<IntentProcessorResult>> ComputeIntentAsync(
+            Document priorDocument,
+            TextSpan priorSelection,
+            Document currentDocument,
+            string? serializedIntentData,
+            CancellationToken cancellationToken)
+        {
+            using var _ = ArrayBuilder<CodeAction>.GetInstance(out var actions);
+            await ComputeRefactoringsAsync(
+                priorDocument,
+                priorSelection,
+                (singleAction, applicableToSpan) => actions.Add(singleAction),
+                (multipleActions) => actions.AddRange(multipleActions),
+                cancellationToken).ConfigureAwait(false);
+
+            if (actions.IsEmpty())
+            {
+                return ImmutableArray<IntentProcessorResult>.Empty;
+            }
+
+            // The refactorings returned will be in the following order (if available)
+            // FieldDelegatingCodeAction, ConstructorDelegatingCodeAction, GenerateConstructorWithDialogCodeAction
+            using var resultsBuilder = ArrayBuilder<IntentProcessorResult>.GetInstance(out var results);
+            foreach (var action in actions)
+            {
+                var intentResult = await GetIntentProcessorResultAsync(action, cancellationToken).ConfigureAwait(false);
+                results.AddIfNotNull(intentResult);
+            }
+
+            return results.ToImmutable();
+
+            static async Task<IntentProcessorResult?> GetIntentProcessorResultAsync(CodeAction codeAction, CancellationToken cancellationToken)
+            {
+                var title = codeAction.Title;
+                var operations = await GetCodeActionOperationsAsync(codeAction, cancellationToken).ConfigureAwait(false);
+
+                // Generate ctor will only return an ApplyChangesOperation or potentially document navigation actions.
+                // We can only return edits, so we only care about the ApplyChangesOperation.
+                var applyChangesOperation = operations.OfType<ApplyChangesOperation>().SingleOrDefault();
+                if (applyChangesOperation == null)
+                {
+                    return null;
+                }
+
+                var type = typeof(CodeAction);
+                return new IntentProcessorResult(applyChangesOperation.ChangedSolution, title, type.Name);
```

Comment:
so *Someone* in the stack should actually ensure that we do not offer changes that have effects beyond just editing the current document.  it probably should be at a higher level so that not all intent providers have to do that determination.

Reply:
In the intent processor I just drop everything that is not for the current document.  Should I instead throw if we have changes outside of a single document?  It seems like we will eventually support more than the single document though